tastebuddies/__init__.py

```diff
@@ -0,0 +1,76 @@
+from pyramid.config import Configurator
+from sqlalchemy import engine_from_config
+
+import os
+
+from pyramid.authentication import AuthTktAuthenticationPolicy
+from pyramid.authorization import ACLAuthorizationPolicy
+from cryptacular.bcrypt import BCRYPTPasswordManager
+
+from .models import (
+    DBSession,
+    Base,
+)
+
+from security import groupfinder, Root, UserFactory
+from pyramid.view import forbidden_view_config
+
+
+def main(global_config, **settings):
+    """ This function returns a Pyramid WSGI application.
+    """
+
+    @forbidden_view_config(renderer='templates/forbidden.jinja2')
+    def forbidden(request):
+        return {}
+
+    engine = engine_from_config(settings, 'sqlalchemy.')
+    DBSession.configure(bind=engine)
+    Base.metadata.bind = engine
+
+    auth_secret = os.environ.get('TASTEBUDDIES_AUTH_SECRET', 'tastynoms')
+
+    authn_policy = AuthTktAuthenticationPolicy(
+        secret=auth_secret,
+        hashalg='sha512',
+        callback=groupfinder,
+    )
+
+    authz_policy = ACLAuthorizationPolicy()
+
+    settings['auth.username'] = os.environ.get('AUTH_USERNAME', 'admin')
+    manager = BCRYPTPasswordManager()
+    settings['auth.password'] = os.environ.get(
+        'AUTH_PASSWORD', manager.encode('secret')
+    )
+
+    config = Configurator(
+        settings=settings,
+        authentication_policy=authn_policy,
+        authorization_policy=authz_policy,
+    )
+
+    config.include('pyramid_jinja2')
+    config.include('pyramid_tm')
+
+    config.add_static_view('static', 'static', cache_max_age=3600)
+
+    config.add_route('home', '/')
+    config.add_route('user_create', '/create_user')
+    config.add_route('user_login', '/login')
+    config.add_route('verify', '/verify', factory=Root)
+    config.add_route('send_email', '/send_email', factory=Root)
+    config.add_route('logout', '/logout', factory=Root)
+    config.add_route('profile_edit', '/profile/edit/{username}',
+                     factory=Root)
+    config.add_route('group_create', '/group/create_group',
+                     factory=Root)
+    config.add_route('group_edit', '/group/edit/{group_name}',
+                     factory=Root)
+    config.add_route('profile_detail', '/profile/{username}',
+                     factory=UserFactory, traverse='/{username}')
+    config.add_route('group_detail', '/group/{group_name}',
+                     factory=Root)
```

Comment:
Think of the spaces made by your URLs and consider whether you can clean them up a bit.  Consistency will help your users to understand the space your app covers more easily.